spec/lib/shanty/plugin_dsl_spec.rb

```diff
@@ -40,4 +49,18 @@
       expect(subject.instance_variable_get(:@tags)).to contain_exactly(:plugin_name, :bar, :lux)
     end
   end
+
+  describe('.provides_config') do
+    it('can set a default value for a config key') do
+      subject.provides_config(:nic, 'cage')
+
+      expect(subject.config[:nic]).to eql('cage')
+    end
+
+    it('can set set an expectation for a config key') do
+      subject.provides_config(:nic)
+
+      expect(subject.config[:nic]).to be_nil
```

Comment:
I feel like this test should have some other kind of assertion that matches the description - perhaps checking that the `:nic` key is stored despite being given no default.